emily/handler/src/database/accessors.rs

```diff
@@ -696,6 +696,69 @@ pub async fn set_api_state(context: &EmilyContext, api_state: &ApiStateEntry) ->
 
 // Limits ----------------------------------------------------------------------
 
+/// Returns height of oldest stacks block anchored to given bitcoin block
+async fn get_oldest_stacks_block_for_bitcoin_block(
+    context: &EmilyContext,
+    bitcoin_height: u64,
+) -> Result<u64, Error> {
+    query_with_partition_key::<ChainstateByBitcoinHeightTableSecondaryIndex>(
+        context,
+        &bitcoin_height,
+        None,
+        None,
+    )
+    .await?
+    .0
+    .iter()
+    .min_by_key(|entry| entry.key.height)
+    .map(|entry| entry.key.height)
+    .ok_or(Error::NotFound)
+}
+
+async fn calculate_sbtc_left_for_withdrawals(
+    context: &EmilyContext,
+    rolling_withdrawal_blocks: Option<u64>,
+    rolling_withdrawal_cap: Option<u64>,
+) -> Result<u64, Error> {
+    if rolling_withdrawal_blocks.is_none() || rolling_withdrawal_cap.is_none() {
+        return Err(Error::NotFound);
+    }
+    let chaintip = get_api_state(context).await?.chaintip();
+    let bitcoin_tip = chaintip.bitcoin_height;
+    if bitcoin_tip.is_none() {
+        return Err(Error::NotFound);
+    }
+    let bitcoin_tip = bitcoin_tip.unwrap();
+    let bitcoin_end_block =
+        bitcoin_tip.saturating_sub(rolling_withdrawal_blocks.unwrap().saturating_sub(1));
+
+    let minimum_stacks_height_in_window =
+        get_oldest_stacks_block_for_bitcoin_block(context, bitcoin_end_block).await?;
+
+    let all_statuses_except_failed: Vec<_> = ALL_STATUSES
+        .iter()
+        .filter(|status| **status != Status::Failed)
+        .collect();
+
+    let mut total_withdrawn = 0u64;
+    for status in all_statuses_except_failed {
+        let withdrawals = get_all_withdrawal_entries_modified_from_height_with_status(
+            context,
+            status,
+            minimum_stacks_height_in_window,
+            None,
+        )
+        .await?;
+        total_withdrawn += withdrawals
+            .iter()
+            .map(|withdrawal| withdrawal.amount)
+            .sum::<u64>();
```

Comment:
You probably want this to be a saturating add. So something like this
```suggestion
        for withdrawal in withdrawals {
            total_withdrawn = total_withdrawn.saturating_add(withdrawal.amount);
        }
```